### PR TITLE
Updated Readme and upload env yaml file

### PR DESCRIPTION
**Pull request by RaghulPS32**
Add readme for training and inference
Added yaml file for environment creation

### CODE REVIEW

**Review by svkeerthy — APPROVED** [review]

File: llvm/lib/CodeGen/MLRegAlloc/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+This repo contains the source code and relevant information described in the [paper](https://dl.acm.org/doi/abs/10.1145/3578360.3580273) ([arXiv](https://arxiv.org/abs/2204.02013)).
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl4real/) for more details.
+
+> RL4ReAl: Reinforcement Learning for Register Allocation, S. VenkataKeerthy, Siddharth Jain, Anilava Kundu, Rohit Aggarwal, Albert Cohen and Ramakrishna Upadrasta
+
+## Environment Setup
+
+
+- Setup the environment using the `.yml` using the following commands
```
Comment (sid18996) [inline llvm/lib/CodeGen/MLRegAlloc/README.md]: remove the listing from description

File: llvm/lib/CodeGen/MLRegAlloc/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    ```bash
+    conda env create -f rl4real_env.yml
+    ```
+
+## Pre-Trained Model Runners for Inference
+
+- Currently three different model runners are supported my the `MLCompilerBridge`
```
Comment (sid18996) [inline llvm/lib/CodeGen/MLRegAlloc/README.md]: 1. Write something like, "There are three modes for inference supported via MLCompilerBridge"
2. Remove the pre-trained from title.
3. mention gRPC based

File: llvm/lib/CodeGen/MLRegAlloc/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    - Query the gRPC Server:
+        - Execute the following `clang/clang++` command to query the server
+            ```bash
+            {BUILD_DIR}/bin/clang -O3 -mllvm -regalloc=greedy -mllvm -mlra-inference -mllvm -ml-config-path=<config path> -mllvm -mlra-server-address=0.0.0.0:<address> <input_data_file>
+            ```
+        - `BUILD_DIR`: Directory where the project is built
+        - `config-path`: Specify the path to the config directory
+        - `input_data_file`: Specify path to  source (`.c/.cpp/.ll`) files
+        - `address`: specify the port gRPC server in running on
```
Comment (sid18996) [inline llvm/lib/CodeGen/MLRegAlloc/README.md]: change address to port

File: llvm/lib/CodeGen/MLRegAlloc/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    - This process starts the  communication between the compiler and the RL model.
+
+
+    ## Inference flow using Pipe Model Runner
+
+    - RL4ReAl supports using Pipe model runner with both `json` and `bytes` data format for communication.
```
Comment (sid18996) [inline llvm/lib/CodeGen/MLRegAlloc/README.md]: Rewrite this sentence

File: model/RL4ReAl/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+This repo contains the source code and relevant information described in the [paper](https://dl.acm.org/doi/abs/10.1145/3578360.3580273) ([arXiv](https://arxiv.org/abs/2204.02013)).
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl4real/) for more details.
+
+> RL4ReAl: Reinforcement Learning for Register Allocation, S. VenkataKeerthy, Siddharth Jain, Anilava Kundu, Rohit Aggarwal, Albert Cohen and Ramakrishna Upadrasta
+
+
+## Environment Setup
+
+- Copy the environment `.yml` files from   `model/RL4ReAl/src/rl4real_env.yml` to the home directory
```
Comment (sid18996) [inline model/RL4ReAl/README.md]: copying file is not needed

File: model/RL4ReAl/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    ```
+## Setup Environment Variables
+
+- Create a `.env` file in the path `model\RL4Real\src`.
+- The `.env` file contains the necessary environment variables.
+- Refer `.env.example` present in `model\RL4Real\src` for setting the required variables.
+
+- `MODEL_DIR`= <path/to/model/dir>
+- `BUILD_DIR`= <path/to/build/dir>
+- `MODEL_PATH`= <path/to/model/file>
```
Comment (sid18996) [inline model/RL4ReAl/README.md]: it should be path to model checkpoint

File: model/RL4ReAl/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+
+    ## Training the Model
+    - Activate the `rllib_env_2.2.0` environment.
+    - Then run the following command
+        ```bash
+        python experiment_ppo.py 
+        ```
+    - Parameters for training should be configured by setting the variables in `model/RL4ReAl/src/ppo_new.py`
+    <!-- - Key parameters to be adjusted: -->
```
Comment (sid18996) [inline model/RL4ReAl/README.md:56]: Uncomment this

**Pull request merged**